src/windows.rs

```diff
@@ -289,37 +286,25 @@ impl WinCredential {
         F: FnOnce(&CREDENTIALW) -> Result<T>,
     {
         self.validate_attributes(None, None)?;
-        let mut p_credential = MaybeUninit::uninit();
+        let mut p_credential = std::ptr::null_mut();
         // at this point, p_credential is just a pointer to nowhere.
         // The allocation happens in the `CredReadW` call below.
         let result = {
             let cred_type = CRED_TYPE_GENERIC;
             let target_name = to_wstr(&self.target_name);
-            unsafe {
-                CredReadW(
-                    target_name.as_ptr(),
-                    cred_type,
-                    0,
-                    p_credential.as_mut_ptr(),
-                )
-            }
+            unsafe { CredReadW(target_name.as_ptr(), cred_type, 0, &mut p_credential) }
         };
         match result {
             0 => {
                 // `CredReadW` failed, so no allocation has been done, so no free needs to be done
                 Err(decode_error())
             }
             _ => {
-                // `CredReadW` succeeded, so p_credential points at an allocated credential.
-                // To do anything with it, we need to cast it to the right type.  That takes two steps:
-                // first we remove the "uninitialized" guard around it, then we reinterpret it as a
-                // pointer to the right structure type.
-                let p_credential = unsafe { p_credential.assume_init() };
-                let w_credential: CREDENTIALW = unsafe { *p_credential };
```

Comment:
This line was making a bitwise copy of the `CREDENTIALW` struct from the heap to the stack, which might not've been intended?